### PR TITLE
Add CI job for running Kubernetes external-storage e2e tests

### PR DESCRIPTION
**Pull request by nixpanic**
This adds a CI job for running the Kubernetes external-storage e2e tests against a PR. The job can be triggered with `/test ci/centos/k8s-e2e-external-storage`. There will be an attempt to test RBD and CephFS, of which RBD is expected to succeed, whereas CephFS seems to have some failure (probably in the driver configuration, _**TO BE TESTED**_).

This PR alone is not sufficient to run the tests, see #1526 for the additions in the master branch.

Merging this PR should not affect any other testing, the job is not started for all PRs, only when requested with a comment.

### CODE REVIEW

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:30]: can this be `rbd.csi.ceph.com`

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:35]: do we have any maximum supported size for rbd?

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:9]: can you add a link to how this is configured? it might be useful if we want to add more cases

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:70]: we do support expansion I think we need to be set to true

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:80]: we support RWX for block mode do we have an option to set it?

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:85]: This is supported

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:89]: This is also supported

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:101]: cephfs supports snapshot don't we need to enable it?

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [inline run-k8s-external-storage-e2e.sh:168]: where this yaml is present.

**Review by Madhu-1 — COMMENTED** [review]
mostly questions

File: run-k8s-external-storage-e2e.sh
Comment (nixpanic) [reply run-k8s-external-storage-e2e.sh:30]: I think this is just a name that gets displayed by the test, but sure, I can change it.

File: run-k8s-external-storage-e2e.sh
Comment (nixpanic) [reply run-k8s-external-storage-e2e.sh:35]: I have no idea how this influences the tests. It probably is limited by the storage that is available in the cluster. It comes from some example file.

File: run-k8s-external-storage-e2e.sh
Comment (nixpanic) [reply run-k8s-external-storage-e2e.sh:9]: yes, I really need to do that. Documentation is very limited though - https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/external is all I have

File: run-k8s-external-storage-e2e.sh
Comment (nixpanic) [reply run-k8s-external-storage-e2e.sh:70]: Sure, but I just hope to get something that passes first. I am considering to place these `driver.yaml` files in the master branch instead of in the ci/centos one. That makes it possible to add features and have them tested with a single PR.

File: run-k8s-external-storage-e2e.sh
Comment (nixpanic) [reply run-k8s-external-storage-e2e.sh:80]: Yes, but enabling RWX also tests filesystem mode as RWX and that fails. I have not found an option to only enable RWX for block.

File: run-k8s-external-storage-e2e.sh
Comment (nixpanic) [reply run-k8s-external-storage-e2e.sh:85]: yeah, but this file is based on initial testing with OCP and OCS where the functionality is not yet part of a release.

File: run-k8s-external-storage-e2e.sh
Comment (nixpanic) [reply run-k8s-external-storage-e2e.sh:168]: it just got generated above

File: run-k8s-external-storage-e2e.sh
Comment (Madhu-1) [reply run-k8s-external-storage-e2e.sh:70]: Yes sounds good to add a Job and move the yaml file to the master branch.

File: run-k8s-external-storage-e2e.sh
Comment (nixpanic) [reply run-k8s-external-storage-e2e.sh:80]: There is also an other issue, see kubernetes/kubernetes#94882 for the bug against the test-case.

**Comment by nixpanic** [comment]
After this is merged, #1526 can be tested by leaving a comment

    /test ci/centos/k8s-e2e-external-storage

**Comment by nixpanic** [comment]
> mostly questions

The files with the description of the drivers and their capabilities have been moved to #1526. These are files with minimal functionality for testing the CI job. Enabling additional functions can be done at a later time.

File: k8s-e2e-external-storage.groovy
Comment (Madhu-1) [inline k8s-e2e-external-storage.groovy:96]: do we need this step @nixpanic?

Comment (nixpanic) [reply k8s-e2e-external-storage.groovy:96]: no, indeed! we use the k8s e2e test that has been built already

File: k8s-e2e-external-storage.groovy
Comment (Madhu-1) [inline k8s-e2e-external-storage.groovy:106]: after this step, we need to update `ceph-csi-config` to have proper clusterID mon details

Comment (nixpanic) [reply k8s-e2e-external-storage.groovy:106]: Right, I guess the proper location would be a script in [`scripts/k8s-storage/`](https://github.com/ceph/ceph-csi/tree/master/scripts/k8s-storage). Do you agree?

Comment (Madhu-1) [reply k8s-e2e-external-storage.groovy:106]: sounds good.

Comment (nixpanic) [reply k8s-e2e-external-storage.groovy:106]: Done, see #1622

**Comment by nixpanic** [comment]
@Yuggupta27 @Madhu-1 could you review and approve this? Once merged, the job is not active, but can be triggered with

    /test ci/centos/k8s-e2e-external-storage

Thanks!

File: k8s-e2e-external-storage.groovy
Comment (Madhu-1) [inline k8s-e2e-external-storage.groovy:29]: @nixpanic still need to update scripts to add GitHub token to avoid rate limit issue?

Comment (nixpanic) [reply k8s-e2e-external-storage.groovy:29]: Oh, yes, forgot about that!

**Comment by nixpanic** [comment]
> /retest ci/centos/jjb-validate

This mostly fails if there is an other jjb-validate job running, see #1624 for a fix

File: k8s-e2e-external-storage.groovy
Comment (Madhu-1) [inline k8s-e2e-external-storage.groovy:111]: helm install-cephcsi creates a configmap better to delete and recreate it again or else this will fail

Comment (nixpanic) [reply k8s-e2e-external-storage.groovy:116]: ok, but that is probably better handled in the create-configmap.sh script

**Review by Madhu-1 — APPROVED** [review]

File: jobs/k8s-e2e-external-storage.yaml
Comment (Yuggupta27) [inline jobs/k8s-e2e-external-storage.yaml:15]: One small question, Do we wish to keep the default as 1.19.3 for now as it was having some issues with CI?

Comment (nixpanic) [reply jobs/k8s-e2e-external-storage.yaml:15]: uhm, right, that is meant to be 1.19.2

**Review by Yuggupta27 — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**
